### PR TITLE
fix(@nguniversal/builders): strip out newlines

### LINKED ISSUE #3121: console.log on the server displays extra new lines
- [ ] builders
- [x ] common  // I guess !
- [ ] express-engine

### Is this a regression?

No, always this way.

### Description

Using console.log on the server shows extra new lines. Even though I call console.log multiple times right next to each other there's often a new line between them.

## 🔬 Minimal Reproduction

1. clone repo https://github.com/tomscript/sturdy-octo-spork
2. npm i -f
3. npm run dev:ssr
4. http://localhost:4200

image

```
    console.log("foo");
    console.log("bar");
    console.log("baz");
```

## 🔥 Exception or Error

none

## 🌍 Your Environment

<pre><code>
Angular CLI: 16.0.0
Node: 18.12.0
Package Manager: npm 8.19.3
OS: darwin arm64

Angular: 16.0.0
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, material, platform-browser, platform-browser-dynamic
... platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1600.0-next.7
@angular-devkit/build-angular   16.0.0-next.7
@angular-devkit/core            16.0.0-next.7
@angular-devkit/schematics      16.0.0
@angular/fire                   7.5.0
@nguniversal/builders           16.0.0
@nguniversal/express-engine     16.0.0
@schematics/angular             16.0.0
rxjs                            7.8.0
typescript                      5.0.4
</code></pre>

### PR DESCRIPTION
**Pull request by tomscript**
Closes #3121

### CODE REVIEW

**Comment by tomscript** [comment]
Hi this is my first contribution. I'm a Googler if that changes anything but I signed the CLA as an individual.
- this is an attempt to fix my own issue i created recently https://github.com/angular/universal/issues/3121

tested by running `bazel build //modules/builders:builders` and just pasting that into `user2/node_modules/@nguniversal/builders/src/ssr-dev-server/index.js`
- also re: unit tests, i didn't know an easy way to test a private func so i exported log. not sure the best way to do that really
- tests pass 

```
bazel test //modules/builders:unit_test --test_output=all
[...]
//modules/builders:unit_test                                             PASSED in 125.8s
```

before

```
[auction.service.getAuction] 736ms from Firestore

[auction.service.getOpenItemsFromFirestore] 1250ms from backend fetch

[auction.service.getItems] 1250ms to get 100 open items (backend)

[app.constructor] Saving injected to TransferState: foo_appspot_com
[app.constructor] sid: foo_appspot_com

[path.service] @Inject(SID) foo_appspot_com

[auction.service] @Inject(SID) foo_appspot_com

[auction.service.getHtml] foo_appspot_com, skip getHtml
```

after

```
[app.config] (server only) environment.sid: foo_appspot_com
[app.config] environment.backend: http://localhost:8080
** Angular Universal Live Development Server is listening on http://localhost:4200, open your browser on http://localhost:4200 **
[app.constructor] Saving injected to TransferState: foo_appspot_com
[app.constructor] sid: foo_appspot_com
[path.service] @Inject(SID) foo_appspot_com
[settings.service] @Inject(SID) foo_appspot_com
[auction.service] @Inject(SID) foo_appspot_com
[auction.service.getHtml] Pull header/footer from env
Angular is running in development mode.
[settings.service] @Inject(SID) foo_appspot_com
[auction.service] @Inject(SID) foo_appspot_com
[auctions-resolver.resolver]
```

**Review by alan-agius4 — DISMISSED** [review]
Hey @tomscript, thanks for this. 

LGTM, just one optional NIT.

File: modules/builders/src/ssr-dev-server/index.ts [hunk 1]
```diff
@@ -173,6 +173,22 @@ export function execute(
...
+  { stderr, stdout }: { stderr: string | undefined; stdout: string | undefined },
+  logger: logging.LoggerApi,
+) {
+  if (stderr) {
+    // Strip the webpack scheme (webpack://) from error log.
+    logger.error(stderr.replace(/\n+$/, '').replace(/webpack:\/\//g, '.'));
+  }
+
+  if (stdout && !IGNORED_STDOUT_MESSAGES.some((x) => stdout.includes(x))) {
+    logger.info(stdout.replace(/\n+$/, ''));
```
Comment (alan-agius4) [inline modules/builders/src/ssr-dev-server/index.ts:188]: Same as above.

**Comment by tomscript** [comment]
oh dang, my commit message was bad, i guess it should have been `buiders: only strip out the last new line, add test`
- er how do i fix that now? or i wonder if that is an optional thing, either way ill happily fix if you can lmk what the best practice here is
- sorry for the noise there

**Comment by alan-agius4** [comment]
@tomscript, you can just squash the commit into the first one.

**Review by alan-agius4 — APPROVED** [review]
LGTM, after fixing the commit issue.

**Comment by tomscript** [comment]
cool, good to know about the squash, will be aware of the future. thanks again @alan-agius4. so i assume you do the submitting here, lmk if ya need anything else from me

**Comment by alan-agius4** [comment]
Thanks again @tomscript.

I’ll merge this tomorrow morning.

**Pull request merged**